### PR TITLE
Define option to lint on load

### PR DESCRIPTION
**Pull request by Hiestaa**
There is no ticket created on this repo, but this links to the ticket in the original plugin repo: https://github.com/TrySound/rollup-plugin-eslint/issues/42

This will cover cases where other plugins would transform the code and cause reported line/character numbers to be inaccurate. Notably, this solves a problem when integrating with typescript which will do a transformation of the code before the eslint plugin passes the code to eslint (including when the eslint plugin is added first in the list of rollup plugins).

### CODE REVIEW

**Comment by rbnts** [comment]
When looking at this issue, also in the context of the original plugin, one question pops up for me: Why not use the `load` hook always? Since according to the [Build Hooks chapter](https://rollupjs.org/guide/en/#build-hooks) of the rollup.js docs, there's different kinds of hooks, some of which can be forced to come `first`. Checking out the [`transform` hook](https://rollupjs.org/guide/en/#transform) in a bit more detail, however, I see that this isn't the case for it. On the other hand, the [`load` hook](https://rollupjs.org/guide/en/#load) comes before the `transform` one, anyway, and could even be utilized in a way to go `first`.

So in that vain, what would you think about directly switching out the `transform` for the `load` hook? Would there be any repercussions? Also pinging @rollup, since I simply don't know.

**Comment by Hiestaa** [comment]
The reason I didn't go ahead with replacing the hook entirely, is I didn't want to break any existing workflow with a change I don't fully understand the consequences of.

However, this repo is pretty new and we could take that opportunity to revisit the choices made by the maintainers of the older one. Looking further into the doc, it looks like the `transform` hook is `sequential`, which means that

> If several plugins implement this hook, all of them will be run in the specified plugin order. If a hook is async, subsequent hooks of this kind will wait until the current hook is resolved.

I have tried to set eslint as first element of the plugins array, but it appears the code provided even to the first transform plugin is already transformed. It could be that another plugin that I used was already applying some transformations in the `load` hook it plugs into.

From that point of view, I think it makes sense to use the `load` hook always. Simplify the configuration a bit, and I don't see any case where a transformed version of the code passed to eslint would be useful. I'll update my PR to get rid of the option and make that the default.

File: src/index.js [hunk 1]
```diff
@@ -15,11 +15,10 @@ module.exports = (options = {}) => {
...
   return {
     name: 'eslint',
-    transform: async (code, id) => {
+    load: async id => {
       if (filter(id)) {
-        const results = await eslint.lintText(code, {
-          filePath: id
-        });
+        const results = await eslint.lintFiles(id);
+
```
Comment (rbnts) [inline src/index.js]:
```suggestion
```

File: src/index.js [hunk 2]
```diff
@@ -43,7 +42,6 @@ module.exports = (options = {}) => {
 
         return result.output || null;
       }
-
```
Comment (rbnts) [inline src/index.js]: Please re-add the line-break here

**Review by rbnts — CHANGES_REQUESTED** [review]
Please remove `package-lock.json` from the PR, the package is using `yarn` instead of `npm`, meaning it already has a `yarn.lock` lockfile.

**Comment by Hiestaa** [comment]
Isn't the choice of the package manager up to the user, rather than the package developper? 

I added it because it seemed to me somebody coming with `npm` might try to install dependencies without using yarn, and having a package-lock in there means it'll work with the dependencies we want. I didn't have any issue getting the package to work with `npm`.

But it's your package, i'll remove if you don't think that point is valid.

**Comment by Hiestaa** [comment]
Actually, it makes sense if we want to have it to add it in a separate PR, keeping this one small. It's gone now.

**Comment by rbnts** [comment]
The choice of the package manager is up to the maintainer of the package. This should have no effect on the end-user of the package, independently of whether they use `yarn` or `npm`.

**Comment by rbnts** [comment]
Dope! Have you ran the tests successfully? I haven't had the time yet to set up CI.

**Comment by Hiestaa** [comment]
yep! 

```
 ~/d/p/rollup-plugin-eslint   feature/lint-on-load  npm run test                                                                                                             3040ms  Fri Aug 28 14:52:43 2020

> @rbnlffl/rollup-plugin-eslint@1.1.1 test /Users/rguyot/dev/perso/rollup-plugin-eslint
> ava test/test.js --verbose


  ✔ rollup runs with the plugin and a configuration (1.2s)
  ✔ rollup runs with the plugin (1.2s)
  ✔ eslint ignores node_modules (1.2s)
  ✔ eslint autofix works (1.2s)
  ─

  4 tests passed
```

I think it's good to go. 

You might wanna consider [`husky`](https://www.npmjs.com/package/husky) to setup a `pre-commit` hook as an alternative to a CI. The guarantee isn't as strong since maintainers can bypass the hook, but it's way easier to setup (and potentially cheaper tho I'm sure there is some free CI available for open source projects). It'll make so that by default the test suite is run on every commit and commits are rejected if it doesn't pass.

**Comment by rbnts** [comment]
Sweet! [`husky`'s already in place](https://github.com/robinloeffel/rollup-plugin-eslint/blob/master/package.json#L45), by the way.

I'll merge your PR, thanks so much for contributing! :)

**Comment by Hiestaa** [comment]
No worries! Thank you for being quick at reviewing, I need that for the workflow for one of my project.

I should really have used `yarn` instead of `npm` - follow the README...  Good work on this 👍

**Comment by rbnts** [comment]
Sweet, let me do some further testing and then I'll release a new version!

**Pull request merged**